manage/upgrading/version_specific_migration/upgrade_to_python3.rst

```diff
@@ -21,11 +21,175 @@ Migrating Plone 5.2 to Python 3
+Check this `Cheat Sheet <http://python-future.org/compatible_idioms.html>`_  with idioms
+for writing Python 2-3 compatible code.
+
+For certain fixes **python-modernize** adds an import of the compatibility library **six** to your code.
+If this happens,
+you might need to fix the order of imports in your file.
+Check the `Python Styleguide for Plone <https://docs.plone.org/develop/styleguide/python.html#grouping-and-sorting>`_
+for information about the order of imports.
+
+Installation
+~~~~~~~~~~~~
+
+Install `modernize <https://pypi.python.org/pypi/modernize>`_ into your Python 3 environment with **pip**.
+
+.. code-block:: shell
+
+    ./bin/pip install modernize
+
+Usage
+~~~~~
+
+The following command runs on fixer on all python files.
+
+.. code-block:: shell
+
+    ./bin/python-modernize -x libmodernize.fixes.fix_import  src/collective.package
+
+.. note::
+
+    The *-x* option is used to exclude certain fixers.
+    The one that adds `from __future__ import absolute_import` should not be used.
+    See ``./bin/python-modernize -l`` for a complete list of fixers and
+    the `Documentation <https://python-modernize.readthedocs.io/en/latest/fixers.html>`_ about them.
+
+The following command applies all fixes the the files in-place:
+
+.. code-block:: shell
+
+    $ ./bin/python-modernize -wn -x libmodernize.fixes.fix_import  src/collective.package
+
+After you ran the command above you can start to tweak what **modernizer** did not get right.
+
+3. Use precompile
+-----------------
+
+You can make use of `plone.recipe.precompiler <https://github.com/plone/plone.recipe.precompiler>`_ to identify syntax errors quickly.
+This recipe compiles all python code already at buildout-time, not at run-time.
+You will see right away, when there is some illegal syntax.
+
+Add the following line(s) to the section [buildout] in  **local.cfg** and
+run `./bin/buildout -c local.cfg` to enable and use **precompile**.
+
+.. code-block:: ini
+
+    parts +=
+    precompiler
+
+4. Start the instance
+---------------------
+
+As a next step we recommend that you try to start the instance with your Addon.
+If it works and you can install the Addon,
+you can some prelimenary manual testing to check for big, obvious issues.
+
+5. Run tests
+------------
+
+.. code-block:: shell
+
+    $ ./bin/test --all -s collective.package
+
+Hopefully there are not many issues with the code left at this point.
+Here is a list of helpful references on the topic of porting Python 2 to Python 3.
+
+    - https://portingguide.readthedocs.io/en/latest/index.html
+    - https://eev.ee/blog/2016/07/31/python-faq-how-do-i-port-to-python-3/
+    - http://www.diveintopython3.net/porting-code-to-python-3-with-2to3.html
+    - https://docs.djangoproject.com/en/1.11/topics/python3/
+    - http://docs.ansible.com/ansible/latest/dev_guide/developing_python3.html
+    - https://docs.python.org/2/library/doctest.html#debugging
+
+TBD: Run tests on travis for testing matrix 2.7, 3.6, 3.7
```

Comment:
In case you don't have a setup for 3.7, have a look at Pyramid's [.travis.yml](https://github.com/Pylons/pyramid/blob/master/.travis.yml#L25-L34). This also tests on py38 with allowed failures. We use [tox](https://github.com/Pylons/pyramid/blob/master/tox.ini) instead of buildout, but you can figure it out.

Reply:
thanks